athenz/src/main/java/com/linecorp/armeria/server/athenz/resource/AthenzResourceProvider.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package com.linecorp.armeria.server.athenz.resource;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+@UnstableApi
+@FunctionalInterface
+public interface AthenzResourceProvider {
+
+    CompletableFuture<String> provide(ServiceRequestContext ctx, HttpRequest req);
```

Comment:
Minor Question) It seems like the overall pattern is providing resources. Is there no need to also determine the action dynamically?

Reply:
From my experience using Athenz, actions are often mapped to HTTP methods (GET, PUT, POST, DELETE, etc.), while resources correspond to paths. In many cases, we simply allow all actions using * and perform authorization primarily based on resources. For that reason, this PR adds only the resource for now.

If we later find a need to handle actions more granularly, we can introduce an action provider at that time.